idaes/generic_models/properties/core/eos/tests/test_ideal.py

```diff
@@ -234,7 +234,7 @@ def test_dens_mol_phase_liq(m):
 @pytest.mark.unit
 def test_dens_mol_phase_vap(m):
     assert str(Ideal.dens_mol_phase(m.props[1], "Vap")) == (
-            'props[1].pressure/(kg*m**2/J/s**2*(8.314462618*(J)/mol/K)'
+            'props[1].pressure/('+str(Ideal.gas_constant(m.props[1])) +
```

Comment:
I did consider point 1, but it was a question of how much effort that would be to implement - here I just went for the quickest fix. Of course ,the whole reason for this fix is because the formatting of the string changed slightly in a newer version of Pyomo, which does pretty much sum up your concern.

Reply:
Yeah.  I think 1 is pretty easy, but if you have a large number of these it may not be worth the effort to update them now.

Reply:
@michaelbynum advocates using `pyomo.core.exprcompare.compare_expressions()` to compare two expressions in tests.

Reply:
@jsiirola That could be useful for the future.